audioread/__init__.py

```diff
@@ -59,9 +59,21 @@ def _mad_available():
     else:
         return True
 
+# A place to store available backends
+BACKENDS = []
 
-def available_backends():
-    """Returns a list of backends that are available on this system."""
+def available_backends(flush_cache=False):
+    """Returns a list of backends that are available on this system.
+
+    The list of backends is cached after the first call.
+    If the parameter `flush_cache` is set to `True`, then the cache
+    will be flushed and the backend list will be reconstructed.
+    """
+    if flush_cache:
+        BACKENDS.clear()
+
+    if BACKENDS:
+        return BACKENDS
```

Comment:
This might be a little shorter and clearer (coupled with the next suggestion below to avoid the need for `clear()`):

```py
if BACKENDS and not flush_cache:
    return BACKENDS
```